### ISSUE TITLE
Extra spaces with text elements when `spaces=number`

### ISSUE THREAD

**Issue by DenisCarriere** [nashwaan/xml-js#5]
I'm getting extra white spaces when I'm using the `text` elements when going two elements deep with `compact=false`.

```javascript
const Elements = {
  type: 'element',
  name: 'a',
  elements: [{
    type: 'element',
    name: 'b',
    elements: [{type: 'text', text: 'foo bar'}],
  }]
}
const xml = convert.js2xml({ elements: [Elements] }, { compact: false, spaces: 4 })
console.log(xml)
```

xml output (notice 4 extra spaces after **foo bar**)
```xml
<a>
    <b>foo bar    </b>
</a>
```

Strange because when using `compact=true` those spaces don't appear.

```javascript
const Elements = {
  a: {
    b: {
      _text: 'foo bar',
    }
  }
}
const xml = convert.js2xml(Elements, { compact: true, spaces: 4 })
console.log(xml)
```
xml output
```xml
<a>
    <b>foo bar</b>
</a>
```

**Comment by nashwaan** [comment]
@DenisCarriere Thanks for reporting this issue. This is now resolved in v0.9.6.
Let me know if the problem still persists.

**Comment by DenisCarriere** [comment]
It works now! 🎉
